### PR TITLE
Travian alliance

### PR DESCRIPTION
**Pull request by RomaMocherniuk**

### CODE REVIEW

**Review by starovoit777 — APPROVED** [review]

File: src/main/java/ua/cv/tim/service/impl/AttackServiceImpl.java [hunk 1]
```diff
@@ -22,43 +27,38 @@
 @Transactional
 public class AttackServiceImpl implements AttackService {
 
-    private static List<Attack> attacks = new ArrayList<>();
+    @Autowired
+    AttackDao attackDao;
+
+    private static final Logger logger = LoggerFactory.getLogger(AttackServiceImpl.class);
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/service/impl/AttackServiceImpl.java:33]: static variables must be declared on top

File: src/main/java/ua/cv/tim/controller/ForgotPassController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        User user = userService.getByMail(email);
+//        User user = new User();
+
+        if (user != null){
+            System.out.println("Password = " + user.getPassword());
+            model.addAttribute("email_send", "Your login and password send to e-mail");
+            userService.sendEmail(user, "Your login is: " + user.getLogin() + ", your password is: " + user.getPassword());
+            return "login.jsp";
+        } else{
+            model.addAttribute("error", "User whith hte same e-mail isn't in DB");
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/controller/ForgotPassController.java:42]: Message - awesome

File: src/main/java/ua/cv/tim/controller/HelpController.java [hunk 3]
```diff
@@ -66,6 +66,7 @@
 
         attack.setPlayerId(id);
         logger.info("add new ask help. Attack: {}", attack);
+        System.out.println(attack);
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/controller/HelpController.java:69]: System.out?

File: src/main/java/ua/cv/tim/dao/hibernate/AttackDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        List<Attack> attacks = (List<Attack>) query.list();
+        return attacks;
+    }
+
+    @Override
+    public List<Attack> getActive() {
+        Session session = getCurrentSession();
+        Query query = session.createQuery("select a FROM Attack a where a.attackTime>:attackTime");
+        Date date = new Date();
+        System.out.println(date);
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/dao/hibernate/AttackDaoImpl.java:32]: again?

File: src/test/java/ua/cv/tim/parser/VillageParserTest.java [hunk 5]
```diff
@@ -32,33 +25,33 @@
...
-        village.setName("1s222i");
-        villageParser.sync(village, "1");
-    }
+//    @Test
+//    public void testPars() throws Exception {
+//        HtmlUnitDriver driver = new HtmlUnitDriver();
+//        driver.setJavascriptEnabled(true);
+//
+//        driver.get("http://ts1.travian.ru/login.php");
+//        driver.findElements(By.name("name")).clear();
```
Comment (ViacheslavKruhlov) [inline src/test/java/ua/cv/tim/parser/VillageParserTest.java:34]: comments in master?!

Comment (RomaMocherniuk) [reply src/test/java/ua/cv/tim/parser/VillageParserTest.java:34]: This tests not passed, so Sergiy said comment them. Maybe better delete them. I will do it.

File: src/main/java/ua/cv/tim/service/impl/AttackServiceImpl.java (hunk 1 above)
Comment (RomaMocherniuk) [reply src/main/java/ua/cv/tim/service/impl/AttackServiceImpl.java:33]: Ok.

File: src/main/java/ua/cv/tim/controller/ForgotPassController.java (hunk 2 above)
Comment (RomaMocherniuk) [reply src/main/java/ua/cv/tim/controller/ForgotPassController.java:42]: I changed logic of reset passqord and message to. I will merge it tomorrow

File: src/main/java/ua/cv/tim/controller/HelpController.java (hunk 3 above)
Comment (RomaMocherniuk) [reply src/main/java/ua/cv/tim/controller/HelpController.java:69]: I have to use loger. Thx. I will delete all System.out

File: src/main/java/ua/cv/tim/dao/hibernate/AttackDaoImpl.java (hunk 4 above)
Comment (RomaMocherniuk) [reply src/main/java/ua/cv/tim/dao/hibernate/AttackDaoImpl.java:32]: Again

**Pull request merged**